src/player.ts

```diff
@@ -96,209 +74,48 @@ export class Player {
-        }
-        if (wallCollision.right) {
-          this.x = this.canvas.width - this.width / 2;
-        } else {
-          this.x += this.diagonalSpeed;
-        }
-      }
-    } else if (keysPressed['ArrowDown'] && keysPressed['ArrowLeft']) {
-      this.facing = FacingAngles.BottomLeft;
-      if (enemyCollision) {
-        this.y -= this.diagonalSpeed * 2;
-        this.x += this.diagonalSpeed * 2;
-        this.takeDamage(10, true);
-      } else {
-        if (wallCollision.bottom) {
-          this.y = this.canvas.height - this.height / 2;
-        } else {
-          this.y += this.diagonalSpeed;
-        }
-        if (wallCollision.left) {
-          this.x = this.width / 2;
-        } else {
-          this.x -= this.diagonalSpeed;
-        }
-      }
-    } else if (keysPressed['ArrowDown'] && keysPressed['ArrowRight']) {
-      this.facing = FacingAngles.BottomRight;
-      if (enemyCollision) {
-        this.y -= this.diagonalSpeed * 2;
-        this.x -= this.diagonalSpeed * 2;
-        this.takeDamage(10, true);
-      } else {
-        if (wallCollision.bottom) {
-          this.y = this.canvas.height - this.height / 2;
-        } else {
-          this.y += this.diagonalSpeed;
-        }
-        if (wallCollision.right) {
-          this.x = this.canvas.width - this.width / 2;
-        } else {
-          this.x += this.diagonalSpeed;
-        }
-      }
-    } else if (keysPressed['ArrowUp']) {
-      this.facing = FacingAngles.Top;
-      if (enemyCollision) {
-        this.y += this.speed * 2;
-        this.takeDamage(10, true);
-      } else if (wallCollision.top) {
-        this.y = this.height / 2;
-      } else {
-        this.y -= this.speed;
-      }
-    } else if (keysPressed['ArrowDown']) {
-      this.facing = FacingAngles.Bottom;
-      if (enemyCollision) {
-        this.y -= this.speed * 2;
-        this.takeDamage(10, true);
-      } else if (wallCollision.bottom) {
-        this.y = this.canvas.height - this.height / 2;
-      } else {
-        this.y += this.speed;
-      }
-    } else if (keysPressed['ArrowLeft']) {
-      this.facing = FacingAngles.Left;
-      if (enemyCollision) {
-        this.x += this.speed * 2;
-        this.takeDamage(10, true);
-      } else if (wallCollision.left) {
-        this.x = this.width / 2;
-      } else {
-        this.x -= this.speed;
-      }
-    } else if (keysPressed['ArrowRight']) {
-      this.facing = FacingAngles.Right;
-      if (enemyCollision) {
-        this.x -= this.speed * 2;
-        this.takeDamage(10, true);
-      } else if (wallCollision.right) {
-        this.x = this.canvas.width - this.width / 2;
-      } else {
-        this.x += this.speed;
-      }
-    }
-  }
-
-  handleAttack(keysPressed: { [key: string]: boolean }, enemies: Array<Enemy>) {
-    if (
-      keysPressed['mousedown'] &&
-      Date.now() - this.attackTimer >= this.weapon.cooldown
-    ) {
-      this.attack();
+      AttackHandler.update(this, enemies);
```

Comment:
It's a little confusing that there's logic for attack handling both in playerController and here. Would it be possible to just handle the attack in playerController? Ideally, there would just be the one function (you should merge `handleAttack` and `update` into one function).

Reply:
Done? The attackHandler is still called in both the playerController and player.update. Since I combined the attackHandler functions (so starting an attack and attack updates are both handled by attackHandler), playerController calls the function to initiate an attack and the player.update calls it to continue the attack calculations.